### DISCUSSION TITLE
Questions about high-quality artbook conversion

### DISCUSSION THREAD

**Discussion by Koomazaz** [apprenticeharper/DeDRM_tools#1903] [Q&A] [answered]
I have spent a few hours researching this and trying different paths, but I am to the point where I need some guidance. I do not own any physical Kindle device, so I cannot use the download functionality on the Amazon site.

I purchased the following artbook from Amazon:
https://www.amazon.com/gp/product/B088QLJ2L9/

My goal is to get the full-resolution artbook downloaded so I can convert it and view it on my PC.

Amazon says that the file size is 2613021 KB (or 2.61 GB), but when I download it via Kindle for PC, it downloads what is obviously a highly-compressed version which is 290 MB of azw and 2 MB of azw.res.

If I download it to my Android phone using the latest version of the Kindle for Android app, I wind up getting something much closer to full-resolution: 2.49 GB of KFX files.

I am aware that there are currently issues with KFX files being imported in Calibre. I also realize that there are issues with getting the decryption keys from the Android version of Kindle. I cannot import the KFX-Zip files into Calibre due to a DRM error.

I have tried downgrading Kindle for PC (1.17, 1.23, etc), and I haven't seen it ever attempt to download more than about 290 MB of data in azw format.

I've tried installing Kindle for Android in Bluestacks, and it also downloads about 290 MB, not the full 2.61 GB.

Finally, I've noticed that if I delete and re-download the KFX files on my Android phone, the total file size downloaded varies wildly. As I mentioned, I have had it download as much as 2.49 GB, but I've also had it download 1.97 GB and less than that, even though the app days it's completely downloaded.

Does anyone have any suggestions on how to best go about making an archive copy of this book with the highest-quality images? Thanks!

**Comment by ElleKayEm** [comment]
If you use Kindle for PC 1.26, you should be able to get a KFX format that you can decrypt.

I have also used this for comixology:  https://github.com/Cortys/comic-backup

I don't know how high-res the images will be.  But it's a couple other things to check out.

**Reply by Koomazaz** [reply]
Hey @ElleKayEm, I sure do appreciate the response.

I tried installing Kindle for PC 1.26, and it downloaded the book as a 290 MB azw file.  Is there some way to force it to download KFX?

**Reply by ElleKayEm** [reply]
If you haven't disabled KFX downloads, then it is likely one part of a KFX format book.  Was there also a voucher file?

**Reply by Koomazaz** [reply]
![Kindle for PC](https://user-images.githubusercontent.com/29680301/139321222-e3b43e8e-c54f-4256-b8de-7b86242fae63.jpg)
This is what Kindle for PC 1.26 downloaded.  Totals 285 MB

![android](https://user-images.githubusercontent.com/29680301/139321336-c952c335-f8cf-4096-9475-99224639a45f.jpg)

This is what Kindle for Android downloaded.  1193 files, totaling 2.49 GB.

I don't think I did anything to disable KFX downloads.  I just installed K4PC, logged in, and clicked the book.

**Reply by j-howell** [reply]
Kindle for PC does not support comics in KFX format. Usually the KF8 main file plus resource container adds up to the full size of the comic but it appears that some limitation is reached with this book.

It appears the only way to get full resolution images for this book is by using the Android or iOS app. It isn’t currently possible to remove DRM in those cases, however it may be possible to do something if the images are unencrypted. Reassembling it would be difficult.

**Reply by Koomazaz** [reply]
@j-howell: I would be more than happy to simply extract the images files (as you said, if they are unencrypted).  Are you suggesting there may be a way to extract the image files without removing the DRM from the Android KFX files?

Do you know of a method to get those images, check if they are unencrypted, and decrypt them, if needed?

Thanks!

**Reply by ElleKayEm** [reply]
No way to force KFX if that's not available for Kindle for PC.  Just a way to force non-KFX.

**Reply by j-howell** [reply]
I took a look at a different large comic, Eternals (https://www.amazon.com/gp/product/B08JD8QJK7). The product page shows it to be 903MB. It downloads to Kindle for PC as two files totaling only 357MB. Kindle for Android downloads the comic as 322 files totaling 785MB.

I looked at at the files from the Android app and found that unfortunately all of them are encrypted. So it doesn't look like there will be any way to extract images from that platform.

**Reply by Koomazaz** [reply]
Thanks to both of you for the insight.  Hopefully, the community will be able to figure out how to get around KFX files and also how to extract Android keys again.

**Comment by j-howell** [comment]
>  I do not own any physical Kindle device, so I cannot use the download functionality on the Amazon site.

I forgot to mention that even if you did have a Kindle the download option would probably give you even more compressed version of the comic.

**Comment by Koomazaz** [comment, marked as answer]
To summarize:
1) Downloading the file from Amazon's webpage would probably be a lower-quality file.
2) Kindle for Android or iOS appears to be the only way to get the highest-resolution files for some books.
3) There is no way to force Kindle for PC to download a book as KFX.  But some books do download as KFX.
4) It seems as though some books may have restrictions that prevent KFX from downloading on PC.  The comics we looked at fall into this category.
5) There is currently no way to remove DRM from files pulled from Android or iOS.

I will eagerly await the day when the devs figure out a way to remove DRM from newer KFX files and to be able to extract Android keys again.

**Reply by ElleKayEm** [reply]
In the meantime, the book you get from Kindle for PC might not be bad for conversion and viewing on a tablet.  You might also look at what you can do with Comic Backup since you can view your Amazon comics on Comixology.  I don't know if that would be better or worse than what you can get from Kindle for PC.

**Reply by mcandre** [reply]
The Comixology website was shredded and changed to a redirect to Amazon.

**Reply by ElleKayEm** [reply]
That hadn't happened yet when I made that comment.
